bitswap-monitoring-client/src/prom.rs

```diff
@@ -89,32 +103,50 @@ pub(crate) struct MetricsByMonitorAndCountry {
     pub(crate) num_disconnected: GenericCounter<AtomicU64>,
 }
 
-impl MetricsByMonitorAndCountry {
+#[derive(Debug, Hash, Eq, PartialEq, Clone, Copy)]
+pub enum OriginType {
+    Peer,
+    Gateway,
+}
+
+impl Metrics {
     /// Creates a set of metrics consisting of a few popular countries and the special
     /// error-condition countries, for the given monitor.
     pub(crate) fn create_basic_set(
         monitor_name: &str,
-    ) -> HashMap<String, MetricsByMonitorAndCountry> {
+    ) -> HashMap<(celes::Country, OriginType), Metrics> {
         let mut metrics_by_country = IntoIterator::into_iter([
-            celes::Country::germany(),
-            celes::Country::the_united_states_of_america(),
-            celes::Country::the_netherlands(),
-        ])
-        .map(|c| {
+            (celes::Country::germany(), OriginType::Peer),
```

Comment:
It's more elegant, and less error-prone, to leave the country iterator as it was, and `map` each country to the two metrics it produces. And then finally `flatten` the iterator of.. whatever you choose -- `Vec`s, arrays, ...

I'd go even further and include the `UNKNOWN` and `ERROR` values in the country iterator. That saves a few lines later in the function, and makes things easier to read (since it's obvious that all countries will be added with both `OriginType`s)